src/tests/integration.test.js

```diff
@@ -57,31 +53,28 @@ describe('client', () => {
     await serverListener.close();
```

Comment:
Minor suggestion: I'd move this up to be executed first. Otherwise, if any exceptions are thrown in `afterAll`, the server will stay up and screw up any subsequent test runs. You wouldn't believe how often this causes CI failures when you have tests like this.